
# Differential abundance analysis: functional

```{r load_data_func,comment="", echo=FALSE, message=FALSE, warning=FALSE}
load("data/data.Rdata")
```

```{r gift, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_counts_filt <- genome_counts[genome_counts$genome %in% rownames(genome_gifts),] 
rownames(genome_counts_filt) <- NULL
#Aggregate bundle-level GIFTs into the compound level
GIFTs_elements <- to.elements(genome_gifts,GIFT_db)
GIFTs_elements_filtered <- GIFTs_elements[rownames(GIFTs_elements) %in% genome_counts_filt$genome,]
GIFTs_elements_filtered <- as.data.frame(GIFTs_elements_filtered) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)

#Aggregate element-level GIFTs into the function level
GIFTs_functions <- to.functions(GIFTs_elements_filtered,GIFT_db)

#Aggregate function-level GIFTs into overall Biosynthesis, Degradation and Structural GIFTs
GIFTs_domains <- to.domains(GIFTs_functions,GIFT_db)

#Get community-weighed average GIFTs per sample
genome_counts_row <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% 
  column_to_rownames(., "genome") 
# genome_counts_row <- rownames_to_column(genome_counts_row, "genome")
GIFTs_elements_community <- to.community(GIFTs_elements_filtered,genome_counts_row,GIFT_db)
GIFTs_functions_community <- to.community(GIFTs_functions,genome_counts_row,GIFT_db)
GIFTs_domains_community <- to.community(GIFTs_domains,genome_counts_row,GIFT_db)
```

```{r comunity_elem_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
GIFTs_elements_community %>%
    as.data.frame() %>%
    rownames_to_column(var="sample")%>%
    pivot_longer(!sample,names_to="trait",values_to="gift") %>%
    left_join(sample_metadata, by = join_by(sample == sample)) %>%
    mutate(functionid = substr(trait, 1, 3)) %>%
    mutate(trait = case_when(
      trait %in% GIFT_db$Code_element ~ GIFT_db$Element[match(trait, GIFT_db$Code_element)],
      TRUE ~ trait
    )) %>%
    mutate(functionid = case_when(
      functionid %in% GIFT_db$Code_function ~ GIFT_db$Function[match(functionid, GIFT_db$Code_function)],
      TRUE ~ functionid
    )) %>%
    mutate(trait=factor(trait,levels=unique(GIFT_db$Element))) %>%
    mutate(functionid=factor(functionid,levels=unique(GIFT_db$Function))) %>%
    ggplot(aes(x=sample,y=trait,fill=gift)) +
        geom_tile(colour="white", linewidth=0.2)+
        scale_fill_gradientn(colours=rev(c("#d53e4f", "#f46d43", "#fdae61", "#fee08b", "#e6f598", "#abdda4", "#ddf1da")))+
        facet_grid(functionid ~ region, scales="free",space="free") +
        theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
              axis.text.y = element_text(size=8),
              strip.text.y = element_text(angle = 0)
              ) +
        labs(y="Traits",x="Samples",fill="GIFT")
```
```{r comunity_funct_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
unique_funct_db<- GIFT_db[c(3,4,5)] %>% 
  distinct(Code_function, .keep_all = TRUE)

GIFTs_functions_community %>%
   t() %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Code_function")  %>%
  left_join(.,unique_funct_db[c(1,3)],by = join_by(Code_function == Code_function))  %>%
  select(-Code_function) %>%
  column_to_rownames(., "Function")%>%
   t()  %>%
    as.data.frame() %>%
    rownames_to_column(var="sample") %>%
    pivot_longer(!sample,names_to="trait",values_to="gift") %>%
    left_join(sample_metadata, by = join_by(sample == sample)) %>%
    ggplot(aes(x=trait,y=sample,fill=gift)) +
        geom_tile(colour="white", linewidth=0.2)+
        scale_fill_gradientn(colours=rev(c("#d53e4f", "#f46d43", "#fdae61", "#fee08b", "#e6f598", "#abdda4", "#ddf1da")))+
        facet_grid(region ~ ., scales="free",space="free")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 10),
        axis.text.y = element_text(size=8),
        strip.background = element_blank(),
        strip.text = element_text(size = 12, color="black",face="bold"),
        axis.title = element_text(size = 12, face = "bold"),
        axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background= element_blank()
              ) +
        labs(x="Function", y="Sample",fill="GIFT")
```

```{r comunity_dom_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
GIFTs_domains_community %>%
    as.data.frame() %>%
    rownames_to_column(var="sample") %>%
    pivot_longer(!sample,names_to="trait",values_to="gift") %>%
    left_join(sample_metadata, by = join_by(sample == sample)) %>%
    ggplot(aes(x=trait,y=sample,fill=gift)) +
        geom_tile(colour="white", linewidth=0.2)+
        scale_fill_gradientn(colours=rev(c("#d53e4f", "#f46d43", "#fdae61", "#fee08b", "#e6f598", "#abdda4", "#ddf1da")))+
        facet_grid(region ~ ., scales="free",space="free")+
  theme(axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size=8),
        strip.background = element_blank(),
        strip.text = element_text(size = 12, color="black",face="bold"),
        axis.title = element_text(size = 12, face = "bold"),
        axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background= element_blank()
        ) +
        labs(x="Function", y="Sample",fill="GIFT")
```
# ONLY USE WILCOXON FOR DA FUNC AS OF 08/12/2024!!!!
## Wilcoxon

### Community elements differences:
```{r comunity_elem, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  merge(., sample_metadata[c(1,4)], by="sample") %>%
  arrange(sample)
```
```{r commun_wilcox_elem, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements <- element_gift %>%
    pivot_longer(-c(sample,region), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
    summarise(p_value = wilcox.test(value ~ region)$p.value) %>%
    mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t <- element_gift  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt <- subset(element_gift_t, trait %in% significant_elements$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,4)], by = join_by(sample == sample))

element_gift_filt %>%
  select(-sample)%>%
  group_by(region)  %>%
  summarise(across(everything(), mean))%>%
   t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))
```

```{r elements_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE}
element_gift_names <- element_gift_filt%>%
  select(-region)%>%
   t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  select(-Elements)%>%
  select(Function, everything())%>%
   t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,4)], by = join_by(sample == sample))


colNames <- names(element_gift_names)[2:69]
for(i in colNames){
  plt <- ggplot(element_gift_names, aes(x=region, y=.data[[i]], color = region, fill=region)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank())
print(plt)
}
```
```{r}
elements_means <- element_gift_filt %>%
  select(-sample) %>%
  group_by(region) %>%
  summarise(across(everything(), mean, na.rm = TRUE)) %>% 
  pivot_longer(-region, names_to = "Elements", values_to = "mean_value") %>%  
  pivot_wider(names_from = region, values_from = mean_value) %>%  
  mutate(
    Enriched_in = case_when(
      `Daneborg` > `Ittoqqortoormiit` ~ "Daneborg",  
      `Daneborg` < `Ittoqqortoormiit` ~ "Ittoqqortoormiit"
    )
  )

significant_elements_plot <- significant_elements %>%
  left_join(elements_means %>% select(Elements, Enriched_in), by = c("trait" = "Elements")) %>%
  left_join(uniqueGIFT_db %>% select(Code_element, Domain, Element, Function), by = c("trait" = "Code_element"))
significant_elements_plot <- significant_elements_plot %>%
  select(-Function.y)
significant_elements_plot <- significant_elements_plot %>%
  mutate(Function.x = sub("_.*", "", Function.x))  


significant_elements_daneb <- significant_elements_plot %>%
  filter(Enriched_in == "Daneborg")
significant_elements_ittoq <- significant_elements_plot %>%
  filter(Enriched_in == "Ittoqqortoormiit")
significant_elements_ittoq_15 <- significant_elements_ittoq %>%
  arrange(desc(p_value)) %>%
  slice_head(n = 15)

func_colors <- c(
  "#1F77B4", "#FF7F0E", "#2CA02C", "#D62728", "#9467BD", 
  "#8C564B", "#E377C2", "#7F7F7F", "#BCBD22", "#17BECF", 
  "#F4A582", "#D6604D", "#F1A340", "#B66E41", "#7B4D19", 
  "#9B1B30", "#3B8C4D"
)

func_table <- as.data.frame(unique(significant_elements_plot$Function.x))
colnames(func_table)[1] <- "Function"
names(func_colors) <- func_table$Function

element_func_plot_daneb <- ggplot(significant_elements_daneb, 
                                  aes(x = forcats::fct_reorder(Element, Function.x),
                                      y = -log10(p_adjust),
                                      color = Function.x)) +
  geom_point(size = 4) + 
  geom_hline(yintercept = 0) + 
  coord_flip() +
  scale_color_manual(values = func_colors) +  # Use custom colors here
  theme(
    axis.text = element_text(size = 8, face = "italic"),
    axis.title = element_text(size = 12),
    legend.position = "none", 
    legend.title = element_blank(),
    panel.background = element_blank(),
    panel.grid.major = element_line(size = 0.15, linetype = 'solid', colour = "grey")
  ) +
  xlab(" ") + 
  ylab("-log10(p-value)") + 
  ggtitle("Daneborg") + 
  guides(col = guide_legend("Function"))

element_func_plot_ittoq <- ggplot(significant_elements_ittoq_15, 
                                  aes(x = forcats::fct_reorder(Element, Function.x),
                                      y = -log10(p_adjust),
                                      color = Function.x)) +
  geom_point(size = 4) + 
  geom_hline(yintercept = 0) + 
  coord_flip() +
  scale_color_manual(values = func_colors) +  # Use custom colors here
  theme(
    axis.text = element_text(size = 8, face = "italic"),
    axis.title = element_text(size = 12),
    legend.position = "none", 
    legend.title = element_blank(),
    panel.background = element_blank(),
    panel.grid.major = element_line(size = 0.15, linetype = 'solid', colour = "grey")
  ) +
  xlab(" ") + 
  ylab("-log10(p-value)") + 
  ggtitle("Ittoqqortoormiit") + 
  guides(col = guide_legend("Function"))

functions_in_both_plots <- union(unique(significant_elements_daneb$Function.x), 
                                 unique(significant_elements_ittoq_15$Function.x))
func_colors_filtered <- func_colors[functions_in_both_plots]
legend_only_plot <- ggplot(data = data.frame(Function = functions_in_both_plots), 
                           aes(x = 1, y = Function, color = Function)) +
  geom_point(size = 3) +
  scale_color_manual(values = func_colors_filtered) + 
  theme_void() +
  theme(legend.position = "right", 
        legend.title = element_blank(), 
        legend.text = element_text(size = 10, face = "italic"))
```
```{r div_plot_together_no_func, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(
  element_func_plot_daneb,
  element_func_plot_ittoq,
  legend_only_plot,
  ncol = 3,
  widths = c(3, 3, 2),
  heights = c(1),
  top = "Top 15 enriched functions in each location"
)
```

### Community functions differences

```{r comunity_func, comment="", echo=FALSE, message=FALSE, warning=FALSE}
function_gift <- GIFTs_functions_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  merge(., sample_metadata[c(1,4)], by="sample")
```

```{r commun_wilcox_func, comment="", echo=FALSE, message=FALSE, warning=FALSE}
unique_funct_db<- GIFT_db[c(3,4,5)] %>% 
  distinct(Code_function, .keep_all = TRUE)


significant_functional <- function_gift %>%
    pivot_longer(-c(sample,region), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
    summarise(p_value = wilcox.test(value ~ region)$p.value) %>%
    mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  left_join(.,unique_funct_db[c(1,3)],by = join_by(trait == Code_function))

significant_functional_ordered <- significant_functional %>%
  arrange(desc(p_value))
```
```{r  function_sig, comment="", echo=FALSE, message=FALSE, warning=FALSE}
function_gift_t <- function_gift  %>% 
  select(-region)  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

function_gift_filt <- subset(function_gift_t, trait %in% significant_functional$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,4)], by = join_by(sample == sample))

function_gift_filt %>%
  select(-sample)%>%
  group_by(region)  %>%
  summarise(across(everything(), mean))%>%
   t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Code_function")  %>%
  left_join(.,unique_funct_db[c(1,3)],by = join_by(Code_function == Code_function))
```

```{r function_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE}
function_gift_names <- function_gift_filt%>%
  select(-region)%>%
   t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Code_function")  %>%
  left_join(.,unique_funct_db[c(1,3)],by = join_by(Code_function == Code_function))%>%
  select(-Code_function)%>%
  select(Function, everything())%>%
   t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,4)], by = join_by(sample == sample))


colNames <- names(function_gift_names)[2:10]
for(i in colNames){
  plt <- ggplot(function_gift_names, aes(x=region, y=.data[[i]], color = region, fill=region)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank())
print(plt)
}
```

### Community domains differences

***No differences***

```{r comunity_dom, comment="", echo=FALSE, message=FALSE, warning=FALSE}
domain_gift <- GIFTs_domains_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  merge(., sample_metadata[c(1,4)], by="sample")
```
```{r commun_wilcox_dom, comment="", echo=FALSE, message=FALSE, warning=FALSE}
unique_domain_db<- GIFT_db[c(4)] %>% 
  distinct(Domain, .keep_all = TRUE)

significant_domain <- domain_gift %>%
  pivot_longer(-c(sample, region), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(
    wilcox_results = list(wilcox.test(value ~ region)),
    p_value = wilcox_results[[1]]$p.value,
    W_score = wilcox_results[[1]]$statistic
  ) %>%
  ungroup() %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  left_join(., unique_domain_db[c(1, 1)], by = join_by(trait == Domain))
```
```{r}
domain_gift_t <- domain_gift  %>% 
  select(-region)  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

domain_gift_filt <- subset(domain_gift_t, trait %in% significant_domain$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,4)], by = join_by(sample == sample))

domain_gift_filt %>%
  select(-sample)%>%
  group_by(region)  %>%
  summarise(across(everything(), mean))%>%
   t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Code_function")  %>%
  left_join(.,unique_domain_db[c(1,1)],by = join_by(Code_function == Domain)) %>%
  column_to_rownames(., var = "Code_function")
```
```{r}
colNames <- names(domain_gift_filt)[2:3]
for(i in colNames){
  plt <- ggplot(domain_gift_filt, aes(x=region, y=.data[[i]], color = region, fill=region)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank())
print(plt)
}
```


# DO NOT USE ANYTHING BELOW THIS LINE AS OF 08/12/2024!!!!!!
## ANCOM-BC2

### Element level
```{r phyloseq, comment="", echo=FALSE, message=FALSE, warning=FALSE}
count_phy <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  select(which(!colSums(., na.rm=TRUE) %in% 0)) %>% 
  t() %>%
  otu_table(., taxa_are_rows=T)

sample_info_tab_phy <- sample_metadata%>%
  column_to_rownames(var="sample")%>%
  sample_data()

TAX <- uniqueGIFT_db%>%
    remove_rownames()%>%
  column_to_rownames(var="Code_element")%>%
  as.matrix()%>%
  tax_table()

physeq_function = phyloseq(count_phy, TAX, sample_info_tab_phy)
  
```

```{r ancom_rand_elem, comment="", echo=FALSE, message=FALSE, warning=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_element = ancombc2(data = physeq_function, 
                  assay_name = "counts", 
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "region", #fixed variable(s)
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```

```{r table_ele, comment="", echo=FALSE, message=FALSE, warning=FALSE}
tax <- data.frame(physeq_function@tax_table) %>%
  rownames_to_column(., "taxon")

ancombc_rand_elem <- ancom_rand_output_element$res %>%
  dplyr::select(taxon, lfc_regionIttoqqortoormiit, p_regionIttoqqortoormiit) %>%
  filter(p_regionIttoqqortoormiit < 0.05) %>%
  dplyr::arrange(p_regionIttoqqortoormiit) %>%
  merge(., tax, by="taxon") %>%
  dplyr::arrange(lfc_regionIttoqqortoormiit)

```
```{r ancom_rand_res_elem, echo=FALSE, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table <- ancom_rand_output_element$res %>%
  dplyr::select(taxon, lfc_regionIttoqqortoormiit, p_regionIttoqqortoormiit) %>%
  filter(p_regionIttoqqortoormiit < 0.05) %>%
  dplyr::arrange(p_regionIttoqqortoormiit) %>%
  merge(., tax, by="taxon")
```
```{r ancombc_rand_plot_elem, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ancombc_rand_table%>%
#      mutate(Function=factor(Function,levels=ancombc_rand_table$Function)) %>%
mutate(Color = ifelse(lfc_regionIttoqqortoormiit <0, "Daneborg","Ittoqqortoormiit")) %>%
ggplot(aes(x=forcats::fct_reorder(Function,lfc_regionIttoqqortoormiit), y=lfc_regionIttoqqortoormiit, fill=Color)) + 
  geom_col() +
#  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10, face="bold.italic"),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
          panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                colour = "grey"))+
  xlab("Traits") + 
  ylab("log2FoldChange")
```

### Functional level
```{r phylo_func, comment="", echo=FALSE, message=FALSE, warning=FALSE}
count_phy <- GIFTs_functions_community %>% 
  as.data.frame() %>% 
  select(which(!colSums(., na.rm=TRUE) %in% 0)) %>% 
  t() %>%
  otu_table(., taxa_are_rows=T)

sample_info_tab_phy <- sample_metadata%>%
  column_to_rownames(var="sample")%>%
  sample_data()

TAX <- unique_funct_db%>%
    remove_rownames()%>%
  column_to_rownames(var="Code_function")%>%
  as.matrix()%>%
  tax_table()

physeq_functional_filtered = phyloseq(count_phy, TAX, sample_info_tab_phy)
physeq_functional_filtered_clr <- microbiome::transform(physeq_functional_filtered, 'clr')  
```

```{r ancom_rand_func, comment="", echo=FALSE, message=FALSE, warning=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_function = ancombc2(data = physeq_functional_filtered, 
                  assay_name = "counts", 
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "region", #fixed variable(s)
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```
```{r table_func, comment="", echo=FALSE, message=FALSE, warning=FALSE}
tax <- data.frame(physeq_functional_filtered@tax_table) %>%
  rownames_to_column(., "taxon")

ancombc_rand_func <- ancom_rand_output_function$res %>%
  dplyr::select(taxon, lfc_regionIttoqqortoormiit, p_regionIttoqqortoormiit) %>%
  filter(p_regionIttoqqortoormiit < 0.05) %>%
  dplyr::arrange(p_regionIttoqqortoormiit) %>%
  merge(., tax, by="taxon") %>%
  dplyr::arrange(lfc_regionIttoqqortoormiit)


```

```{r ancom_rand_res_funct, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ancombc_rand_table_func <- ancom_rand_output_function$res %>%
  dplyr::select(taxon, lfc_regionIttoqqortoormiit, p_regionIttoqqortoormiit) %>%
  filter(p_regionIttoqqortoormiit < 0.05) %>%
  dplyr::arrange(p_regionIttoqqortoormiit) %>%
  merge(., tax, by="taxon")
```
```{r ancombc_rand_plot_funct, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=4, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_func%>%
mutate(Color = ifelse(lfc_regionIttoqqortoormiit <0, "Daneborg","Ittoqqortoormiit")) %>%
ggplot(aes(x=forcats::fct_reorder(Function,lfc_regionIttoqqortoormiit), y=lfc_regionIttoqqortoormiit, fill=Color)) + 
  geom_col() +
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10, face="bold.italic"),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
          panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                colour = "grey"))+
  xlab("Traits") + 
  ylab("log2FoldChange")
```