---
title: "ARCHIVE_analyses"
author: "Elsa"
date: "`r Sys.Date()`"
output: html_document
---
04_community_composition
### Genus

```{r taxonomy_genus_summary, warning=FALSE, comments="", message=FALSE}
genus_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  left_join(genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  group_by(sample,phylum,genus) %>%
  summarise(relabun=sum(count)) %>%
  filter(genus != "g__") %>%
  mutate(genus= sub("^g__", "", genus))

genus_summary_sort <- genus_summary %>%
    group_by(genus) %>%
    summarise(mean=mean(relabun, na.rm=T),sd=sd(relabun, na.rm=T)) %>%
    arrange(-mean) 

genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% 
  left_join(sample_metadata, by = join_by(sample == sample)) %>% 
  left_join(genome_metadata, by = join_by(genome == genome)) %>% 
  group_by(sample,phylum,genus, region) %>%
  summarise(relabun=sum(count)) %>%
  filter(genus != "g__") %>%
  mutate(genus= sub("^g__", "", genus))%>%
  group_by(genus) %>%
  summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              High_mean=mean(relabun[region=="Daneborg"]*100, na.rm=T),
             High_sd=sd(relabun[region=="Daneborg"]*100, na.rm=T),
              Low_mean=mean(relabun[region=="Ittoqqortoormiit"]*100, na.rm=T),
              Low_sd=sd(relabun[region=="Ittoqqortoormiit"]*100, na.rm=T)) %>%
  mutate(Total=str_c(round(total_mean,3),"±",round(total_sd,3)),
           Daneborg=str_c(round(High_mean,3),"±",round(High_sd,3)),
           Ittoqqortoormiit=str_c(round(Low_mean,3),"±",round(Low_sd,3))) %>% 
  arrange(-total_mean) %>% 
  dplyr::select(genus,Total,Daneborg,Ittoqqortoormiit) %>% 
  tt()
```

```{r taxonomy_jitterplot_genus, fig.height=14, fig.width=10, fig.fullwidth=TRUE}
genus_arrange <- genus_summary %>%
    group_by(genus) %>%
    summarise(mean=sum(relabun)) %>%
    filter(genus != "g__")%>%
    arrange(-mean) %>%
    select(genus) %>%
    mutate(genus= sub("^g__", "", genus)) %>%
    pull()

#Per region
genus_summary %>%
    left_join(sample_metadata,by=join_by(sample==sample)) %>%
    mutate(genus=factor(genus, levels=rev(genus_summary_sort %>% pull(genus)))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=genus, group=genus, color=phylum)) +
        scale_color_manual(values=phylum_colors) +
        geom_jitter(alpha=0.5) + 
        facet_grid(.~region)+
        theme_minimal() + 
        labs(y="Family", x="Relative abundance", color="Phylum")

```

07_differential_abundance_taxonomy

```{r ancombc_rand_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ggplot(ancombc_rand_table, aes(x=forcats::fct_rev(genus), y=lfc_regionIttoqqortoormiit, color=phylum)) + 
  geom_point(size=4) + 
  scale_color_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10, face="bold.italic"),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
          panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                colour = "grey"))+
  xlab("Genus") + 
  ylab("log2FoldChange")+
  guides(col=guide_legend("Phylum"))
```
```{r ancombc_rand_plot_top_15, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ggplot(ancombc_rand_table, aes(x=forcats::fct_rev(genus), y=lfc_regionIttoqqortoormiit, color=phylum)) + 
  geom_point(size=4) + 
  scale_color_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10, face="bold.italic"),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
          panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                colour = "grey"))+
  xlab("Genus") + 
  ylab("log2FoldChange")+
  guides(col=guide_legend("Phylum"))
```
```{r}
top_20_mags_ittoq <- ancombc_rand_table %>%
  arrange(desc(lfc_regionIttoqqortoormiit)) %>%
  slice_head(n = 20)
top_20_mags_ittoq <- top_20_mags_ittoq %>%
  mutate(
    name = if_else(
      species == "s__",  # Check if species is exactly "s__"
      paste(genus, "spp.", sep = " "),  # Replace with genus + "spp."
      str_remove(species, "^s__")  # Remove the "s__" prefix from known species
    )
  )
tax_table <- as.data.frame(unique(top_20_mags_ittoq$phylum))
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
plot1_DA<- ggplot(top_20_mags_ittoq, aes(x = forcats::fct_reorder(name, lfc_regionIttoqqortoormiit), y = lfc_regionIttoqqortoormiit, color = phylum, )) + 
  geom_point(size = 4) + 
  scale_color_manual(values = tax_color) + 
  geom_hline(yintercept = 0) + 
  coord_flip() +
  theme(
    axis.text = element_text(size = 8, face = "italic"),
    axis.title = element_text(size = 12),
    legend.position = "right", 
    legend.title = element_blank(),
    panel.background = element_blank(),
    panel.grid.major = element_line(size = 0.15, linetype = 'solid', colour = "grey")
  ) +
  xlab(" ") + 
  ylab("log2FoldChange") +
  ggtitle("Enriched in Ittoqqortoormiit") + 
  guides(col = guide_legend("Phylum"))
```
```{r}
top_20_mags_daneb <- ancombc_rand_table %>%
  arrange(lfc_regionIttoqqortoormiit) %>%  # Sort by the lowest values
  slice_head(n = 20)  # Select the first 20 rows
top_20_mags_daneb <- top_20_mags_daneb %>%
  mutate(
    name = if_else(
      species == "s__",  # Check if species is exactly "s__"
      paste(genus, "spp.", sep = " "),  # Replace with genus + "spp."
      str_remove(species, "^s__")  # Remove the "s__" prefix from known species
    )
  )
tax_table <- as.data.frame(unique(top_20_mags_daneb$phylum))
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()

plot2_DA<- ggplot(top_20_mags_daneb, aes(x = forcats::fct_reorder(name, lfc_regionIttoqqortoormiit), y = lfc_regionIttoqqortoormiit, color = phylum, )) + 
  geom_point(size = 4) + 
  scale_color_manual(values = tax_color) + 
  geom_hline(yintercept = 0) + 
  coord_flip() +
  theme(
    axis.text = element_text(size = 8, face = "italic"),
    axis.title = element_text(size = 12),
    legend.position = "none", 
    legend.title = element_blank(),
    panel.background = element_blank(),
    panel.grid.major = element_line(size = 0.15, linetype = 'solid', colour = "grey")
  ) +
  xlab(" ") + 
  ylab("log2FoldChange") +
  ggtitle("Enriched in Daneborg") +  
  guides(col = guide_legend("Phylum"))
```
```{r div_plot_together_no_func, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(plot2_DA,plot1_DA,ncol = 2)) 
```
## Ancombc2 (considering structural zeros): Genus level

```{r name_gen, comment="", echo=FALSE,message=FALSE, warning=FALSE}
physeq_genome_filtered_test<- physeq_genome_filtered
tax <- data.frame(tax_table(physeq_genome_filtered_test))

tax.clean <- data.frame(row.names = row.names(tax),
domain = str_replace(tax[,1], "d__",""),
phylum = str_replace(tax[,2], "D_1__",""),
class = str_replace(tax[,3], "c__",""),
order = str_replace(tax[,4], "o__",""),
family = str_replace(tax[,5], "f__",""),
genus = str_replace(tax[,6], "g__",""),
species = str_replace(tax[,7], "s__",""),
stringsAsFactors = FALSE)
tax.clean[is.na(tax.clean)] <- ""

for (i in 1:7){ tax.clean[,i] <- as.character(tax.clean[,i])}
####### Fille holes in the tax table
tax.clean[is.na(tax.clean)] <- ""
for (i in 1:nrow(tax.clean)){

#Fill in missing taxonomy
if (tax.clean[i,2] == ""){
domain <- paste("domain_", tax.clean[i,1], sep = "")
tax.clean[i, 2:7] <- domain
} else if (tax.clean[i,3] == ""){
phylum <- paste("phylum_", tax.clean[i,2], sep = "")
tax.clean[i, 3:7] <- phylum
} else if (tax.clean[i,4] == ""){
class <- paste("class_", tax.clean[i,3], sep = "")
tax.clean[i, 4:7] <- class
} else if (tax.clean[i,5] == ""){
order <- paste("order_", tax.clean[i,4], sep = "")
tax.clean[i, 5:7] <- order
} else if (tax.clean[i,6] == ""){
family <- paste("family_", tax.clean[i,5], sep = "")
tax.clean[i, 6:7] <- family
} else if (tax.clean[i,7] == ""){
tax.clean$species[i] <- paste("genus",tax.clean$genus[i], sep = "_")
}
}

tax_table(physeq_genome_filtered_test) <- as.matrix(tax.clean)
```

```{r ancom_rand_gen, comment="", echo=FALSE,message=FALSE, warning=FALSE}
ancom_rand_output_gen = ancombc2(data = physeq_genome_filtered_test, 
                  assay_name = "counts",
                  tax_level = "genus", #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "region", #fixed variable(s)
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0, 
                  s0_perc = 0,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = NULL,
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r ancom_rand_res_gen, comment="", echo=FALSE,message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(phylum, genus), ~ str_replace(., "[dpcofgs]__", ""))%>%
  select(phylum, genus)%>%
	unique()

ancombc_rand_table_gen <- ancom_rand_output_gen$res %>%
  mutate_at(vars(taxon), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::select(taxon, lfc_regionIttoqqortoormiit, p_regionIttoqqortoormiit) %>%
  filter(p_regionIttoqqortoormiit < 0.05) %>%
  dplyr::arrange(lfc_regionIttoqqortoormiit)  %>%
  left_join(taxonomy, by=join_by(taxon == genus)) %>%
  dplyr::arrange(lfc_regionIttoqqortoormiit)

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_gen$phylum))

colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_gen, comment="", echo=FALSE,message=FALSE, warning=FALSE}
ancombc_rand_table_gen%>%
      mutate(taxon=factor(taxon,levels=ancombc_rand_table_gen$taxon)) %>%
mutate(Color = ifelse(lfc_regionIttoqqortoormiit <0, "Daneborg","Ittoqqortoormiit")) %>%
ggplot(., aes(x=lfc_regionIttoqqortoormiit, y=forcats::fct_rev(taxon), fill=Color)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Genera")+
  guides(fill=guide_legend(title="Group"))

```

```{r ancom_rand_volcano_gen, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ancom_result_gen <- ancom_rand_output_gen$res %>%
  na.omit() %>%
  dplyr::rename(genome=taxon) %>%
  left_join(genome_metadata,by=join_by(genome==genome))
ancom_result_gen %>%
    mutate(significance = ifelse(p_regionIttoqqortoormiit < 0.05, "1", "0")) %>%
    ggplot(., aes(x=-log(p_regionIttoqqortoormiit), y=lfc_regionIttoqqortoormiit, color=significance)) +
      geom_point() +
      scale_color_manual(values = c("#cccccc","#00FFFF")) +
      geom_text(aes(3, 5), label = "Enriched\nin Ittoqqortoormiit", color="#666666") +
      geom_text(aes(3, -5), label = "Enriched\nin Danenborg", color="#666666") +
      labs(color="Significance", y="Difference between locations", x="p-value") +
      theme_classic()
```

# Phylum level:
```{r ancom_rand_gen, comment="", echo=FALSE,message=FALSE, warning=FALSE}
ancom_rand_output_phylum = ancombc2(data = physeq_genome_filtered_test, 
                  assay_name = "counts",
                  tax_level = "phylum", #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "region", #fixed variable(s)
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0, 
                  s0_perc = 0,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = NULL,
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```
```{r ancom_rand_res_gen, comment="", echo=FALSE,message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))%>%
  select(phylum)%>%
	unique()

ancombc_rand_table_phylum <- ancom_rand_output_phylum$res %>%
  mutate_at(vars(taxon), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::select(taxon, lfc_regionIttoqqortoormiit, p_regionIttoqqortoormiit) %>%
  filter(p_regionIttoqqortoormiit < 0.05) %>%
  dplyr::arrange(lfc_regionIttoqqortoormiit)  %>%
  left_join(taxonomy, by=join_by(taxon == phylum)) %>%
  dplyr::arrange(lfc_regionIttoqqortoormiit)

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_phylum$taxon))

colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```
```{r ancombc_rand_plot_phylum, comment="", echo=FALSE,message=FALSE, warning=FALSE}
ancombc_rand_table_phylum%>%
      mutate(taxon=factor(taxon,levels=ancombc_rand_table_phylum$taxon)) %>%
mutate(Color = ifelse(lfc_regionIttoqqortoormiit <0, "Daneborg","Ittoqqortoormiit")) %>%
ggplot(., aes(x=lfc_regionIttoqqortoormiit, y=forcats::fct_rev(taxon), fill=Color)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
#  coord_flip()+
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Phylum")+
  guides(fill=guide_legend(title="Group"))

```
```{r ancom_rand_volcano_phylum, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ancom_result_phylum <- ancom_rand_output_phylum$res %>%
  na.omit() %>%
  dplyr::rename(genome=taxon) %>%
  left_join(genome_metadata,by=join_by(genome==genome))
ancom_result_phylum %>%
    mutate(significance = ifelse(p_regionIttoqqortoormiit < 0.05, "1", "0")) %>%
    ggplot(., aes(x=-log(p_regionIttoqqortoormiit), y=lfc_regionIttoqqortoormiit, color=significance)) +
      geom_point() +
      scale_color_manual(values = c("#cccccc","#00FFFF")) +
      geom_text(aes(3, 5), label = "Enriched\nin Ittoqqortoormiit", color="#666666") +
      geom_text(aes(3, -5), label = "Enriched\nin Danenborg", color="#666666") +
      labs(color="Significance", y="Difference between locations", x="p-value") +
      theme_classic()
```


08_differential_abundance_functional
### Community elements differences:
```{r comunity_elem_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
GIFTs_elements_community %>%
    as.data.frame() %>%
    rownames_to_column(var="sample")%>%
    pivot_longer(!sample,names_to="trait",values_to="gift") %>%
    left_join(sample_metadata, by = join_by(sample == sample)) %>%
    mutate(functionid = substr(trait, 1, 3)) %>%
    mutate(trait = case_when(
      trait %in% GIFT_db$Code_element ~ GIFT_db$Element[match(trait, GIFT_db$Code_element)],
      TRUE ~ trait
    )) %>%
    mutate(functionid = case_when(
      functionid %in% GIFT_db$Code_function ~ GIFT_db$Function[match(functionid, GIFT_db$Code_function)],
      TRUE ~ functionid
    )) %>%
    mutate(trait=factor(trait,levels=unique(GIFT_db$Element))) %>%
    mutate(functionid=factor(functionid,levels=unique(GIFT_db$Function))) %>%
    ggplot(aes(x=sample,y=trait,fill=gift)) +
        geom_tile(colour="white", linewidth=0.2)+
        scale_fill_gradientn(colours=rev(c("#d53e4f", "#f46d43", "#fdae61", "#fee08b", "#e6f598", "#abdda4", "#ddf1da")))+
        facet_grid(functionid ~ region, scales="free",space="free") +
        theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
              axis.text.y = element_text(size=8),
              strip.text.y = element_text(angle = 0)
              ) +
        labs(y="Traits",x="Samples",fill="GIFT")
```
```{r final_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
#element_plot <- 
code_function2 %>%
  left_join(significant_elements, by=join_by(Elements==trait)) %>%
  left_join(log_fold, by=join_by(Elements==elements)) %>% 
  left_join(gift_colors, by=join_by(Code_function==Code_function)) %>% 
  ggplot(., aes(x = logfc_region, y = -log(p_adjust), color=legend, size=abs(Difference))) +
  geom_jitter(width = 0.2, height = 0.2)+
  geom_vline(xintercept=0) +
  scale_color_manual(values = gift_colors$Color)+
  theme_classic()+
  labs(size="Mean difference (abs)", color="Functional trait")+
  labs(x = "Log-fold change", y="-Log adjusted p-value") +
  geom_text_repel(aes(label = Element), min.segment.length = 0.4, size=2.5, max.overlaps = Inf)
```
```{r function_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE}
function_gift_names <- function_gift_filt%>%
  select(-region)%>%
   t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Code_function")  %>%
  left_join(.,unique_funct_db[c(1,3)],by = join_by(Code_function == Code_function))%>%
  select(-Code_function)%>%
  select(Function, everything())%>%
   t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,4)], by = join_by(sample == sample))


colNames <- names(function_gift_names)[2:10]
for(i in colNames){
  plt <- ggplot(function_gift_names, aes(x=region, y=.data[[i]], color = region, fill=region)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank())
print(plt)
}
```
### Community functions differences
```{r comunity_funct_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
unique_funct_db<- GIFT_db[c(3,4,5)] %>% 
  distinct(Code_function, .keep_all = TRUE)

GIFTs_functions_community %>%
   t() %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Code_function")  %>%
  left_join(.,unique_funct_db[c(1,3)],by = join_by(Code_function == Code_function))  %>%
  select(-Code_function) %>%
  column_to_rownames(., "Function")%>%
   t()  %>%
    as.data.frame() %>%
    rownames_to_column(var="sample") %>%
    pivot_longer(!sample,names_to="trait",values_to="gift") %>%
    left_join(sample_metadata, by = join_by(sample == sample)) %>%
    ggplot(aes(x=trait,y=sample,fill=gift)) +
        geom_tile(colour="white", linewidth=0.2)+
        scale_fill_gradientn(colours=rev(c("#d53e4f", "#f46d43", "#fdae61", "#fee08b", "#e6f598", "#abdda4", "#ddf1da")))+
        facet_grid(region ~ ., scales="free",space="free")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 10),
        axis.text.y = element_text(size=8),
        strip.background = element_blank(),
        strip.text = element_text(size = 12, color="black",face="bold"),
        axis.title = element_text(size = 12, face = "bold"),
        axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background= element_blank()
              ) +
        labs(x="Function", y="Sample",fill="GIFT")
```

```{r comunity_func, comment="", echo=FALSE, message=FALSE, warning=FALSE}
function_gift <- GIFTs_functions_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  merge(., sample_metadata[c(1,4)], by="sample")
```

```{r commun_wilcox_func, comment="", echo=FALSE, message=FALSE, warning=FALSE}
unique_funct_db<- GIFT_db[c(3,4,5)] %>% 
  distinct(Code_function, .keep_all = TRUE)


significant_functional <- function_gift %>%
    pivot_longer(-c(sample,region), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
    summarise(p_value = wilcox.test(value ~ region)$p.value) %>%
    mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  left_join(.,unique_funct_db[c(1,3)],by = join_by(trait == Code_function))

significant_functional_ordered <- significant_functional %>%
  arrange(desc(p_value))
```
```{r  function_sig, comment="", echo=FALSE, message=FALSE, warning=FALSE}
function_gift_t <- function_gift  %>% 
  select(-region)  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

function_gift_filt <- subset(function_gift_t, trait %in% significant_functional$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,4)], by = join_by(sample == sample))

function_gift_filt %>%
  select(-sample)%>%
  group_by(region)  %>%
  summarise(across(everything(), mean))%>%
   t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Code_function")  %>%
  left_join(.,unique_funct_db[c(1,3)],by = join_by(Code_function == Code_function))
```
```{r}
sig_function <- significant_functional_ordered %>% 
  select(trait) %>% 
  pull()

GIFTs_functions_community %>% 
  as.data.frame() %>% 
  select(any_of(sig_function)) %>% 
  rownames_to_column(., "sample") %>% 
  left_join(sample_metadata[c(1,4)], by=join_by(sample==sample)) %>% 
  select(-sample) %>% 
  pivot_longer(!region, names_to="trait",values_to="gift") %>% 
  left_join(significant_functional_ordered, by=join_by(trait==trait)) %>% 
  ggplot(aes(x=region, y=gift, color=region)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
  scale_x_discrete(labels=c("Daneborg" = "Daneborg", "Ittoqqortoormiit" = "Ittoqqortoormiit")) +
facet_grid(.~Function,  scales="free_x")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10, angle = 45, hjust = 1),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )
```
### Community domains differences
```{r comunity_dom_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
GIFTs_domains_community %>%
    as.data.frame() %>%
    rownames_to_column(var="sample") %>%
    pivot_longer(!sample,names_to="trait",values_to="gift") %>%
    left_join(sample_metadata, by = join_by(sample == sample)) %>%
    ggplot(aes(x=trait,y=sample,fill=gift)) +
        geom_tile(colour="white", linewidth=0.2)+
        scale_fill_gradientn(colours=rev(c("#d53e4f", "#f46d43", "#fdae61", "#fee08b", "#e6f598", "#abdda4", "#ddf1da")))+
        facet_grid(region ~ ., scales="free",space="free")+
  theme(axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size=8),
        strip.background = element_blank(),
        strip.text = element_text(size = 12, color="black",face="bold"),
        axis.title = element_text(size = 12, face = "bold"),
        axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background= element_blank()
        ) +
        labs(x="Function", y="Sample",fill="GIFT")
```

```{r comunity_dom, comment="", echo=FALSE, message=FALSE, warning=FALSE}
domain_gift <- GIFTs_domains_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  merge(., sample_metadata[c(1,4)], by="sample")
```
```{r commun_wilcox_dom, comment="", echo=FALSE, message=FALSE, warning=FALSE}
unique_domain_db<- GIFT_db[c(4)] %>% 
  distinct(Domain, .keep_all = TRUE)

significant_domain <- domain_gift %>%
  pivot_longer(-c(sample, region), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(
    wilcox_results = list(wilcox.test(value ~ region)),
    p_value = wilcox_results[[1]]$p.value,
    W_score = wilcox_results[[1]]$statistic
  ) %>%
  ungroup() %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  left_join(., unique_domain_db[c(1, 1)], by = join_by(trait == Domain))
```
```{r}
domain_gift_t <- domain_gift  %>% 
  select(-region)  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

domain_gift_filt <- subset(domain_gift_t, trait %in% significant_domain$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,4)], by = join_by(sample == sample))

domain_gift_filt %>%
  select(-sample)%>%
  group_by(region)  %>%
  summarise(across(everything(), mean))%>%
   t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Code_function")  %>%
  left_join(.,unique_domain_db[c(1,1)],by = join_by(Code_function == Domain)) %>%
  column_to_rownames(., var = "Code_function")
```
```{r}
colNames <- names(domain_gift_filt)[2:3]
for(i in colNames){
  plt <- ggplot(domain_gift_filt, aes(x=region, y=.data[[i]], color = region, fill=region)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank())
print(plt)
}
```

## Func capacity of structural zeros
```{r structural_functional_capacity2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=12, fig.width=10, fig.fullwidth=TRUE}
GIFTs_elements_filtered %>% 
  rownames_to_column("genome") %>%
  filter(genome %in% struct_mag)  %>% 
    pivot_longer(!genome,names_to="trait",values_to="gift") %>% 
    left_join(struct_mag_meta, by = join_by(genome == genome))%>%
    mutate(functionid = substr(trait, 1, 3)) %>%
    mutate(trait = case_when(
      trait %in% GIFT_db$Code_element ~ GIFT_db$Element[match(trait, GIFT_db$Code_element)],
      TRUE ~ trait
    )) %>%
    mutate(functionid = case_when(
      functionid %in% GIFT_db$Code_function ~ GIFT_db$Function[match(functionid, GIFT_db$Code_function)],
      TRUE ~ functionid
    )) %>%
    mutate(trait=factor(trait,levels=unique(GIFT_db$Element))) %>%
    mutate(functionid=factor(functionid,levels=unique(GIFT_db$Function))) %>%
    ggplot(aes(x=genome,y=trait,fill=gift)) +
        geom_tile(colour="white", linewidth=0.2)+
        scale_fill_gradientn(colours=rev(c("#d53e4f", "#f46d43", "#fdae61", "#fee08b", "#e6f598", "#abdda4", "#ddf1da")))+
        facet_grid(functionid ~ present, scales="free",space="free") +
        theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
              axis.text.y = element_text(size=6),
              strip.text.y = element_text(angle = 0)
              ) +
        labs(y="Traits",x="Samples",fill="GIFT")

```
## Functional capacity of the structural MAGs
```{r structural_zeros}
ittoq_samples <- sample_metadata %>% 
  filter(region == "Ittoqqortoormiit") %>%
  dplyr::select(sample) %>%
  pull()

daneb_samples <- sample_metadata %>% 
  filter(region == "Daneborg") %>%
  dplyr::select(sample) %>%
  pull()

structural_zeros <- genome_counts %>% 
   rowwise() %>% #compute for each row (genome)
   mutate(all_zeros_ittoq = all(c_across(all_of(ittoq_samples)) == 0)) %>% # set true if all samples in Ittoq have zeros
   mutate(all_zeros_daneb = all(c_across(all_of(daneb_samples)) == 0)) %>% # set true if all samples in Daneborg have zeros
   mutate(average_ittoq = mean(c_across(all_of(ittoq_samples)), na.rm = TRUE)) %>% # get average genome counts across Ittoq
   mutate(average_daneb = mean(c_across(all_of(daneb_samples)), na.rm = TRUE)) %>% # get average genome counts across Daneborg
   filter(all_zeros_ittoq == TRUE || all_zeros_daneb==TRUE)  %>% # filter only genomes with structural zeros
   mutate(present = case_when(
      all_zeros_ittoq & !all_zeros_daneb ~ "Daneborg",
      !all_zeros_ittoq & all_zeros_daneb ~ "Ittoq",
      !all_zeros_ittoq & !all_zeros_daneb ~ "None",
      TRUE ~ NA_character_
    )) %>%
   mutate(average = ifelse(present == "Ittoq", average_ittoq, average_daneb)) %>%
   dplyr::select(genome, present, average) %>%
   left_join(genome_metadata, by=join_by(genome==genome)) %>%
   arrange(present,-average)
structural_zeros
struct_mag <- structural_zeros %>% 
  filter(present %in% c("Ittoq","Daneborg")) %>% 
  select(genome) %>% 
  pull()
struct_mag_meta <- structural_zeros %>% 
  filter(present %in% c("Ittoq","Daneborg"))

# Element
GIFTs_elements_filtered_struc_zeros <- GIFTs_elements[rownames(GIFTs_elements) %in% structural_zeros$genome,]
GIFTs_elements_filtered_struc_zeros <- as.data.frame(GIFTs_elements_filtered_struc_zeros) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)
GIFTs_elements_community_struc_zeros <- to.community(GIFTs_elements_filtered_struc_zeros,genome_counts_row,GIFT_db)

# Functions
GIFTs_functions_filtered_struc_zeros <- GIFTs_functions[rownames(GIFTs_functions) %in% structural_zeros$genome,]
GIFTs_functions_filtered_struc_zeros <- as.data.frame(GIFTs_functions_filtered_struc_zeros) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)
GIFTs_functions_community_struc_zeros <- to.community(GIFTs_functions_filtered_struc_zeros,genome_counts_row,GIFT_db)

# Domains
GIFTs_domains_filtered_struc_zeros <- GIFTs_domains[rownames(GIFTs_domains) %in% structural_zeros$genome,]
GIFTs_domains_filtered_struc_zeros <- as.data.frame(GIFTs_domains_filtered_struc_zeros) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)
GIFTs_domains_community_struc_zeros <- to.community(GIFTs_domains_filtered_struc_zeros,genome_counts_row,GIFT_db)
```

### element level
```{r}
element_gift_struc_zeros <- GIFTs_elements_community_struc_zeros %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  merge(., sample_metadata[c(1,4)], by="sample") %>%
  arrange(sample)
```
```{r commun_wilcox_elem, comment="", echo=FALSE, message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements_struc_zeros <- element_gift_struc_zeros %>%
    pivot_longer(-c(sample,region), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
    summarise(p_value = wilcox.test(value ~ region)$p.value) %>%
    mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t_struc_zeros <- element_gift_struc_zeros  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt_struc_zeros <- subset(element_gift_t_struc_zeros, trait %in% significant_elements_struc_zeros$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,4)], by = join_by(sample == sample))

element_gift_filt_struc_zeros %>%
  select(-sample)%>%
  group_by(region)  %>%
  summarise(across(everything(), mean))%>%
   t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))

difference_table_struc_zeros <- element_gift_filt_struc_zeros %>%
  select(-sample) %>%
  group_by(region) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Ittoqqortoormiit-Daneborg)%>% 
  mutate(group_color = ifelse(Difference <0, "Daneborg","Ittoq")) 
```
```{r}
element_gift_names_struc_zeros <- element_gift_filt_struc_zeros%>%
  select(-region)%>%
   t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(Elements == Code_element))%>%
  select(-Elements)%>%
  select(Function, everything())%>%
   t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,4)], by = join_by(sample == sample))


colNames <- names(element_gift_names_struc_zeros)[2:113]
for(i in colNames){
  plt <- ggplot(element_gift_names_struc_zeros, aes(x=region, y=.data[[i]], color = region, fill=region)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank())
print(plt)
}
```
### function level
```{r comunity_func, comment="", echo=FALSE, message=FALSE, warning=FALSE}
function_gift_struc_zeros <- GIFTs_functions_community_struc_zeros %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  merge(., sample_metadata[c(1,4)], by="sample")
```

```{r commun_wilcox_func, comment="", echo=FALSE, message=FALSE, warning=FALSE}
unique_funct_db<- GIFT_db[c(3,4,5)] %>% 
  distinct(Code_function, .keep_all = TRUE)


significant_functional_struc_zeros <- function_gift_struc_zeros %>%
    pivot_longer(-c(sample,region), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
    summarise(p_value = wilcox.test(value ~ region)$p.value) %>%
    mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  left_join(.,unique_funct_db[c(1,3)],by = join_by(trait == Code_function))

significant_functional_ordered_struc_zeros <- significant_functional_struc_zeros %>%
  arrange(desc(p_value))
```

```{r  function_sig, comment="", echo=FALSE, message=FALSE, warning=FALSE}
function_gift_t_struc_zeros <- function_gift_struc_zeros  %>% 
  select(-region)  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

function_gift_filt_struc_zeros <- subset(function_gift_t_struc_zeros, trait %in% significant_functional_struc_zeros$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,4)], by = join_by(sample == sample))

function_gift_filt_struc_zeros %>%
  select(-sample)%>%
  group_by(region)  %>%
  summarise(across(everything(), mean))%>%
   t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Code_function")  %>%
  left_join(.,unique_funct_db[c(1,3)],by = join_by(Code_function == Code_function))
```
```{r function_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE}
function_gift_names_struc_zeros <- function_gift_filt_struc_zeros%>%
  select(-region)%>%
   t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Code_function")  %>%
  left_join(.,unique_funct_db[c(1,3)],by = join_by(Code_function == Code_function))%>%
  select(-Code_function)%>%
  select(Function, everything())%>%
   t()%>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,4)], by = join_by(sample == sample))


colNames <- names(function_gift_names_struc_zeros)[2:20]
for(i in colNames){
  plt <- ggplot(function_gift_names_struc_zeros, aes(x=region, y=.data[[i]], color = region, fill=region)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank())
print(plt)
}
```
### domain level 
```{r comunity_dom, comment="", echo=FALSE, message=FALSE, warning=FALSE}
domain_gift_struc_zeros <- GIFTs_domains_community_struc_zeros %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  merge(., sample_metadata[c(1,4)], by="sample")
```
```{r commun_wilcox_dom, comment="", echo=FALSE, message=FALSE, warning=FALSE}
unique_domain_db<- GIFT_db[c(4)] %>% 
  distinct(Domain, .keep_all = TRUE)

significant_domain_struc_zeros <- domain_gift_struc_zeros %>%
  pivot_longer(-c(sample, region), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(
    wilcox_results = list(wilcox.test(value ~ region)),
    p_value = wilcox_results[[1]]$p.value,
    W_score = wilcox_results[[1]]$statistic
  ) %>%
  ungroup() %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  left_join(., unique_domain_db[c(1, 1)], by = join_by(trait == Domain))
```
```{r}
domain_gift_t_struc_zeros <- domain_gift_struc_zeros  %>% 
  select(-region)  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

domain_gift_filt_struc_zeros <- subset(domain_gift_t_struc_zeros, trait %in% significant_domain_struc_zeros$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,4)], by = join_by(sample == sample))

domain_gift_filt_struc_zeros %>%
  select(-sample)%>%
  group_by(region)  %>%
  summarise(across(everything(), mean))%>%
   t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "Code_function")  %>%
  left_join(.,unique_domain_db[c(1,1)],by = join_by(Code_function == Domain)) %>%
  column_to_rownames(., var = "Code_function")
```
```{r}
colNames <- names(domain_gift_filt_struc_zeros)[2:4]
for(i in colNames){
  plt <- ggplot(domain_gift_filt_struc_zeros, aes(x=region, y=.data[[i]], color = region, fill=region)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
    scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
    scale_fill_manual(values=c("#e5bd5b", "#6b7398"))+
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank())
print(plt)
}
```

# ############################################################### # ############################################################### # ###############################################################
# DO NOT USE ANYTHING BELOW THIS LINE AS OF 08/12/2024!!!!!!
# ############################################################### # ############################################################### # ###############################################################
## ANCOM-BC2 (DO NOT USE)

### Element level
```{r phyloseq, comment="", echo=FALSE, message=FALSE, warning=FALSE}
count_phy <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  select(which(!colSums(., na.rm=TRUE) %in% 0)) %>% 
  t() %>%
  otu_table(., taxa_are_rows=T)

sample_info_tab_phy <- sample_metadata%>%
  column_to_rownames(var="sample")%>%
  sample_data()

TAX <- uniqueGIFT_db%>%
    remove_rownames()%>%
  column_to_rownames(var="Code_element")%>%
  as.matrix()%>%
  tax_table()

physeq_function = phyloseq(count_phy, TAX, sample_info_tab_phy)
  
```

```{r ancom_rand_elem, comment="", echo=FALSE, message=FALSE, warning=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_element = ancombc2(data = physeq_function, 
                  assay_name = "counts", 
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "region", #fixed variable(s)
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```

```{r table_ele, comment="", echo=FALSE, message=FALSE, warning=FALSE}
tax <- data.frame(physeq_function@tax_table) %>%
  rownames_to_column(., "taxon")

ancombc_rand_elem <- ancom_rand_output_element$res %>%
  dplyr::select(taxon, lfc_regionIttoqqortoormiit, p_regionIttoqqortoormiit) %>%
  filter(p_regionIttoqqortoormiit < 0.05) %>%
  dplyr::arrange(p_regionIttoqqortoormiit) %>%
  merge(., tax, by="taxon") %>%
  dplyr::arrange(lfc_regionIttoqqortoormiit)

```
```{r ancom_rand_res_elem, echo=FALSE, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table <- ancom_rand_output_element$res %>%
  dplyr::select(taxon, lfc_regionIttoqqortoormiit, p_regionIttoqqortoormiit) %>%
  filter(p_regionIttoqqortoormiit < 0.05) %>%
  dplyr::arrange(p_regionIttoqqortoormiit) %>%
  merge(., tax, by="taxon")
```
```{r ancombc_rand_plot_elem, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ancombc_rand_table%>%
#      mutate(Function=factor(Function,levels=ancombc_rand_table$Function)) %>%
mutate(Color = ifelse(lfc_regionIttoqqortoormiit <0, "Daneborg","Ittoqqortoormiit")) %>%
ggplot(aes(x=forcats::fct_reorder(Function,lfc_regionIttoqqortoormiit), y=lfc_regionIttoqqortoormiit, fill=Color)) + 
  geom_col() +
#  geom_point(size=4) + 
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10, face="bold.italic"),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
          panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                colour = "grey"))+
  xlab("Traits") + 
  ylab("log2FoldChange")
```

### Functional level
```{r phylo_func, comment="", echo=FALSE, message=FALSE, warning=FALSE}
count_phy <- GIFTs_functions_community %>% 
  as.data.frame() %>% 
  select(which(!colSums(., na.rm=TRUE) %in% 0)) %>% 
  t() %>%
  otu_table(., taxa_are_rows=T)

sample_info_tab_phy <- sample_metadata%>%
  column_to_rownames(var="sample")%>%
  sample_data()

TAX <- unique_funct_db%>%
    remove_rownames()%>%
  column_to_rownames(var="Code_function")%>%
  as.matrix()%>%
  tax_table()

physeq_functional_filtered = phyloseq(count_phy, TAX, sample_info_tab_phy)
physeq_functional_filtered_clr <- microbiome::transform(physeq_functional_filtered, 'clr')  
```

```{r ancom_rand_func, comment="", echo=FALSE, message=FALSE, warning=FALSE}
set.seed(1234) #set seed for reproducibility
ancom_rand_output_function = ancombc2(data = physeq_functional_filtered, 
                  assay_name = "counts", 
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "region", #fixed variable(s)
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```
```{r table_func, comment="", echo=FALSE, message=FALSE, warning=FALSE}
tax <- data.frame(physeq_functional_filtered@tax_table) %>%
  rownames_to_column(., "taxon")

ancombc_rand_func <- ancom_rand_output_function$res %>%
  dplyr::select(taxon, lfc_regionIttoqqortoormiit, p_regionIttoqqortoormiit) %>%
  filter(p_regionIttoqqortoormiit < 0.05) %>%
  dplyr::arrange(p_regionIttoqqortoormiit) %>%
  merge(., tax, by="taxon") %>%
  dplyr::arrange(lfc_regionIttoqqortoormiit)


```

```{r ancom_rand_res_funct, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ancombc_rand_table_func <- ancom_rand_output_function$res %>%
  dplyr::select(taxon, lfc_regionIttoqqortoormiit, p_regionIttoqqortoormiit) %>%
  filter(p_regionIttoqqortoormiit < 0.05) %>%
  dplyr::arrange(p_regionIttoqqortoormiit) %>%
  merge(., tax, by="taxon")
```
```{r ancombc_rand_plot_funct, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=4, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_func%>%
mutate(Color = ifelse(lfc_regionIttoqqortoormiit <0, "Daneborg","Ittoqqortoormiit")) %>%
ggplot(aes(x=forcats::fct_reorder(Function,lfc_regionIttoqqortoormiit), y=lfc_regionIttoqqortoormiit, fill=Color)) + 
  geom_col() +
  scale_fill_manual(values=c("#e5bd5b", "#6b7398")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10, face="bold.italic"),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
          panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                colour = "grey"))+
  xlab("Traits") + 
  ylab("log2FoldChange")
```

# Maybe delete this one. check tomorrow. 
```{r}

gift_colors <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes)%>% 
  mutate(legend=str_c(Code_function," - ",Function))


elements_means <- element_gift_filt %>%
  select(-sample) %>%
  group_by(region) %>%
  summarise(across(everything(), mean, na.rm = TRUE)) %>% 
  pivot_longer(-region, names_to = "Elements", values_to = "mean_value") %>%  
  pivot_wider(names_from = region, values_from = mean_value) %>%  
  mutate(
    Enriched_in = case_when(
      `Daneborg` > `Ittoqqortoormiit` ~ "Daneborg",  
      `Daneborg` < `Ittoqqortoormiit` ~ "Ittoqqortoormiit"
    )
  )

significant_elements_plot <- significant_elements %>%
  left_join(elements_means %>% select(Elements, Enriched_in), by = c("trait" = "Elements")) %>%
  left_join(uniqueGIFT_db %>% select(Code_element, Domain, Element, Function), by = c("trait" = "Code_element"))
significant_elements_plot <- significant_elements_plot %>%
  select(-Function.y)
significant_elements_plot <- significant_elements_plot %>%
  mutate(Function.x = sub("_.*", "", Function.x))  


significant_elements_daneb <- significant_elements_plot %>%
  filter(Enriched_in == "Daneborg")
significant_elements_ittoq <- significant_elements_plot %>%
  filter(Enriched_in == "Ittoqqortoormiit")
significant_elements_ittoq_15 <- significant_elements_ittoq %>%
  arrange(desc(p_value)) %>%
  slice_head(n = 15)

gift_colors <- read_tsv("data/gift_colors.tsv") %>% 
  filter(Code_function %in% unique_codes)%>% 
  mutate(legend=str_c(Code_function," - ",Function))

func_table <- as.data.frame(unique(significant_elements_plot$Function.x))
colnames(func_table)[1] <- "Function"
names(func_colors) <- func_table$Function

element_func_plot_daneb <- ggplot(significant_elements_daneb, 
                                  aes(x = forcats::fct_reorder(Element, Function.x),
                                      y = -log10(p_adjust),
                                      color = Function.x)) +
  geom_point(size = 4) + 
  geom_hline(yintercept = 0) + 
  coord_flip() +
  scale_color_manual(values = gift_colors) +  # Use custom colors here
  theme(
    axis.text = element_text(size = 8, face = "italic"),
    axis.title = element_text(size = 12),
    legend.position = "none", 
    legend.title = element_blank(),
    panel.background = element_blank(),
    panel.grid.major = element_line(size = 0.15, linetype = 'solid', colour = "grey")
  ) +
  xlab(" ") + 
  ylab("-log10(p-value)") + 
  ggtitle("Daneborg") + 
  guides(col = guide_legend("Function"))

element_func_plot_ittoq <- ggplot(significant_elements_ittoq_15, 
                                  aes(x = forcats::fct_reorder(Element, Function.x),
                                      y = -log10(p_adjust),
                                      color = Function.x)) +
  geom_point(size = 4) + 
  geom_hline(yintercept = 0) + 
  coord_flip() +
  scale_color_manual(values = gift_colors) +  # Use custom colors here
  theme(
    axis.text = element_text(size = 8, face = "italic"),
    axis.title = element_text(size = 12),
    legend.position = "none", 
    legend.title = element_blank(),
    panel.background = element_blank(),
    panel.grid.major = element_line(size = 0.15, linetype = 'solid', colour = "grey")
  ) +
  xlab(" ") + 
  ylab("-log10(p-value)") + 
  ggtitle("Ittoqqortoormiit") + 
  guides(col = guide_legend("Function"))

functions_in_both_plots <- union(unique(significant_elements_daneb$Function.x), 
                                 unique(significant_elements_ittoq_15$Function.x))
func_colors_filtered <- func_colors[functions_in_both_plots]
legend_only_plot <- ggplot(data = data.frame(Function = functions_in_both_plots), 
                           aes(x = 1, y = Function, color = Function)) +
  geom_point(size = 3) +
  scale_color_manual(values = func_colors_filtered) + 
  theme_void() +
  theme(legend.position = "right", 
        legend.title = element_blank(), 
        legend.text = element_text(size = 10, face = "italic"))
```
```{r div_plot_together_no_func, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(
  element_func_plot_daneb,
  element_func_plot_ittoq,
  legend_only_plot,
  ncol = 3,
  widths = c(3, 3, 2),
  heights = c(1),
  top = "Top 15 enriched functions in each location"
)
```
